src/Graphics/OpenGLContext/GLSL/glsl_SpecialShadersFactory.cpp

```diff
@@ -60,20 +60,46 @@ namespace glsl {
 				"uniform lowp usampler2D uTlutImage;\n"
 				"uniform sampler2D uDepthImage;		\n"
 				"uniform lowp vec4 uFogColor;								\n"
-				"OUT lowp vec4 fragColor;									\n"
+				;
+
+			if (config.frameBufferEmulation.N64DepthCompare != 0 && _glinfo.imageTextures)
+				m_part += "layout(binding = 2, r32f) highp uniform restrict readonly image2D uDepthImageZ;		\n";
+
+			if (config.frameBufferEmulation.N64DepthCompare != 0 && _glinfo.ext_fetch) {
+				m_part +=
+					"layout(location = 0) OUT lowp vec4 fragColor;	\n"
+					"layout(location = 1) inout highp vec4 depthZ;	\n"
+					;
+			} else
+				m_part += "OUT lowp vec4 fragColor;									\n";
```

Comment:
I feel dizzy when I'm looking at all these ifs.

Are '_glinfo.imageTextures' and '_glinfo.ext_fetch' mutually exclusive? Would not it be more correct to rewrite as:

			if (config.frameBufferEmulation.N64DepthCompare != 0) {
				if (_glinfo.imageTextures)
					m_part += "layout(binding = 2, r32f) highp uniform restrict readonly image2D uDepthImageZ;		\n";
				
				if (_glinfo.ext_fetch) {
					m_part +=
						"layout(location = 0) OUT lowp vec4 fragColor;	\n"
						"layout(location = 1) inout highp vec4 depthZ;	\n"
						;
				else
					m_part += "OUT lowp vec4 fragColor;									\n";
			} else
				m_part += "OUT lowp vec4 fragColor;									\n";